league.js

```diff
@@ -0,0 +1,323 @@
+                        true
+                    );
+                    if (!date.isValid()) {
+                        date = undefined;
+                    }
+                    new_pairings.push({
+                        white: self.canonical_username(row['white'].value),
+                        black: self.canonical_username(row['black'].value),
+                        result: link['text'],
+                        url: link['href'],
+                        scheduled_date: date
+                    });
+                });
+                self._pairings = new_pairings;
+                callback(undefined, self._pairings);
+            }
+        );
+    },
+
+    //--------------------------------------------------------------------------
+    // Finds the pairing for this current round given either a black or a white
+    // username.
+    //--------------------------------------------------------------------------
+    'findPairing': function(white, black) {
+        white = white || undefined;
+        if (!white) {
+            throw new Error("findPairing requires at least one username.");
+        }
+        black = black || undefined;
+        var possibilities = this._pairings;
+        function filter(playerName) {
+            if (playerName) {
+                possibilities = _.filter(possibilities, function(item) {
+                    return (
+                        item.white.toLowerCase().includes(playerName) ||
+                        item.black.toLowerCase().includes(playerName)
+                    );
+                });
+            }
+        };
+        filter(white);
+        filter(black);
+        return possibilities;
+    },
+    //--------------------------------------------------------------------------
+    // Prepare a debug message for this league
+    //--------------------------------------------------------------------------
+    'debugMessage': function() {
+        var self = this;
+
+        return Q.fcall(function() {
+            return  'DEBUG:\nLeague: {name}\nTotal Pairings: {pairingsCount}\nLast Updated: {lastUpdated} [{since} ago]'.format({
+                name: self.options.name,
+                pairingsCount: self._pairings.length,
+                lastUpdated: self._lastUpdated.format("YYYY-MM-DD HH:mm UTC"),
+                since: self._lastUpdated.fromNow(true)
+            })
+        });
+    },
+    //--------------------------------------------------------------------------
+    // Generates the appropriate data format for pairing result for this league.
+    //--------------------------------------------------------------------------
+    'getPairingDetails': function(targetPlayer) {
+        var self = this;
+        return Q.fcall(function() {
+            var pairings = self.findPairing(targetPlayer.name);
+            if (pairings.length < 1) {
+                return {};
+            }
+            // TODO: determine what to do if multiple pairings are returned. ?
```

Comment:
I suggest Including multiples and telling the caller that they should notify the moderators if this is unexpected.